doc/nyan/base types.md

```diff
@@ -0,0 +1,164 @@
+	# WIP3: if WIP2 is not possible: implement an 'Ability' that allows to gather only one resource.
+	# A worker would then need 4 gather abilities added to it's 'abilities' set.
+	# And this would allow to reference
+
+	# WIP4: how do we reference Animations?
+#	animation : file # one picture that contains all frames or, if a directory,
+	                # contains all frames. # WIP5: naming scheme?
+#	renameMe : file # WIP6: how to read `animation'? describes, where in the big picture `animation' which frames are; whether to potentially mirror them.
+	                # possibilities: 1. $renameMe is a regular file: description of `animation' (syntax?)
+	                # 2. replace `renameMe' by more fields that describe `animation' (ugly imo, but maybe the best way)
+	                # 3. remove `renameMe':
+	                #    if $animation is a directory: look for '$animation/animate.nfo'
+	                #    else: look for "$(basename $animation png)nfo"
+
+CarryAbility(Ability): # WIP: find a better name?
+	# Used by priests for relicts.
+	# WIP1: how should this look like? keep it general? can maybe be used by buildings to let units find shelter in them? or by mods for adding new relict-like objects
+	# WIP2: can maybe merged with a modified GatherAbility?
+
+#	goodMemberName : Portable # what can be collected? make a 'Portable' class and let relict inherit that? could then be also used by resources?
+#	maxAmount
+
+	animation : file # what does a priest look like when he's carrying it?
+	                 # WIP: see GatherAbility.animation and nyan question 4 (further down).
+
+AttackAbility(Ability):
+	# Every attack shall inherit from this.
+	# WIP: how will damage calculation work in oa?
+
+	range : float
+	rate : float
+	damage : float # rename to baseDamage ?
+
+
+Unit(): # WIP
+	# Everything you can see in the game # (WIP: but not assets, which is basically everything you see. Maybe better:)
+	# Everything that has some 'Ability' inherits from this.
+
+	ability : set(Ability)
+
+
+
+#and probably more. if you know what's missing, add it.
+
+```
+
+Open nyan implementation questions:
+
+1. How do we say in Nyan, that a Towncenter attack depends on how many people are inside the building?
+2. What about the building 'Ability' to carry units? will this lead to recursive references?
+3. How do we reference hooks from Nyan/create new hooks in mods?
+
+  -> If the {Towncenter,Castle} has certain damage, contained units need to leave.
+  
+  -> A relict in a Church leads to increasing gold.
+4. How do we add new animations?
+
+  -> How would a mod add a new relict-like object that can be carried (the animation)? can we describe composition of frames? How is it done by the original game?
+  
+  -> We need a unified way/member-name to add a new animation to an 'Ability'
+  
+  -> animations are stored in one big picture (or allow one file per frame?). How do we tell where in the picture is which frame? How can you tell to mirror a frame?
```

Comment:
In the original game, the relic thing is actually a new unit: "monk with relic". This is kinda stupid because it assumes only monks can carry relics, which is find in age2 but very unsuitable for us.

So we can prepare the combination of a `Carryable` and some `Unit` in the sense that the carryable has to define its size and position, and we can then render the carryable in front of the unit. We can even define workaround replacement graphics that way in the sense that if it's a "known combination", the combination image is used (aka the monk with relic animation). How we model this in nyan I cannot tell right now (and haven't thought about it yet, so very good catch here).


The image metainformation should still be done in the good old `.docx` files as they're currently describing where a subimage is in one image. Things like animation speed, looping etc is defined in nyan though.
The engine will provide a nyan object `Sprite` which has a `file` member and others like the speed, etc. When this sprite is then required to be shown (e.g. because a unit uses it as its image) the engine will handle everything else internally, like parsing the subtexture information from the `.docx`, drawing it etc.